src/convertToObject.js

```diff
@@ -13,6 +13,17 @@
  */
 function convertToObject(sourceString) {
   // write your code here
+  return sourceString
+    .split(';')
+    .filter(field => field.trim().length > 2)
+    .map(keyValue => keyValue.split(':').map(prop => prop.trim()))
+    .reduce((object, property) => {
+      const keyValue = {};
+
+      keyValue[property[0]] = property[1];
+
+      return Object.assign(object, keyValue);
```

Comment:
FYI: You can make new object right in assign method. Of course it's better to do it when object isn't big and complex.

Reply:
Good point! Thanks)